org.eclipse.scanning.event/src/org/eclipse/scanning/event/QueueReader.java

```diff
@@ -134,15 +134,15 @@ public QueueReader(IEventConnectorService service, Comparator<? super T> compara
 	}
 
 	/**
-	 * 
+	 *
 	 * @param uri
 	 * @param topicName
 	 * @param clazz
 	 * @param monitorTime
 	 * @return
 	 */
 	public Map<String, T> getHeartbeats(final URI uri, final String topicName, final Class<T> clazz, final long monitorTime) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)